### PR TITLE
Fit to raw checkbox now works as intended

### LINKED ISSUE #32187: Muon guis tries to fit rebinned data when not present
**Describe the bug**
Muon gui tries to fit rebinned data when not present
**To Reproduce**

1. Open muon analysis gui
2. load current run
3.  Try unchecking fit to raw checkbox this should give a warning 
4. Try unchecking fit to raw checkbox again and it unchecks checkbox which it should not

**Expected behavior**
Fit to raw checkbox should raise a warning when checkbox is unchecked with no rebinned data present 

**Platform/Version (please complete the following information):**
 - OS: [e.g. Windows, RHEL 7, Ubuntu, macOS]
 - OS Version:
 - Mantid Version [e.g. 6.1.0]

### PR DESCRIPTION
**Pull request by tolu28-coder**
**Description of work.**
This pull request fixes the isssue with the fit to raw checkbox in muon guis unchecking when no rebinned  workspaces are present.

**To test:**
1. Open muon analysis gui
2. load current run
3.  Try unchecking fit to raw checkbox this should give a warning 
4. Trying to uncheck the fit to raw checkbox again should give a warning no matter how many time you try when no rebined data is present.

Fixes #32187. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
I just looked on master and it works as expected for all runs, except the current run. I think we should be investigating why it doesn't work for that run instead of applying a sticking plaster.

**Comment by tolu28-coder** [comment]
The issue still occurs on other runs i try what dataset are you using

**Comment by AnthonyLim23** [comment]
I using MUSR on master, 62260 is fine and so is the one before the current run

**Comment by tolu28-coder** [comment]
The issue still occurs for me no matter what run i use

**Comment by AnthonyLim23** [comment]
Then I think you may have a bad build. 
image

**Comment by tolu28-coder** [comment]
> Then I think you may have a bad build.
> image

Did you try unchecking checkbox twice

File: scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_function_options_view.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from Muon.GUI.Common.utilities import table_utils
 
 from qtpy.QtWidgets import QWidget
+from qtpy import QtCore
```
Comment (robertapplin) [inline scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_function_options_view.py:15]: Can you move this above the `QWidget` import, and also only import `Qt` as this is all we need from `QtCore`
```suggestion
from qtpy.QtCore import Qt
from qtpy.QtWidgets import QWidget
```

**Review by robertapplin — CHANGES_REQUESTED** [review]
I can confirm that this is a good fix to the problem you found, I have just one small suggestion

**Comment by robertapplin** [comment]
I can replicate this bug for all runs

I think `setChecked` doesn't update the `Qt.CheckState` and only makes a visual change to the check box, hence why it was not previously working with the `QCheckbox::stateChanged` signal

**Comment by robertapplin** [comment]
It looks like something has gone wrong with the merge

File: scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_function_options_view.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from Muon.GUI.Common.utilities import table_utils
 
 from qtpy.QtWidgets import QWidget
+from qtpy.QtCore import Qt
```
Comment (robertapplin) [inline scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_function_options_view.py:15]: Thanks for changing this, can you also put this import above the `qtpy.QtWidgets` import?

**Review by robertapplin — CHANGES_REQUESTED** [review]

**Comment by tolu28-coder** [comment]
I was able to fix it by resetting branch and rebasing on master

**Review by robertapplin — DISMISSED** [review]

**Review by robertapplin — APPROVED** [review]

**Pull request merged**